tests/appsec/iast/test_iast_no_request.py

```diff
@@ -0,0 +1,33 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2021 Datadog, Inc.
+
+from utils import BaseTestCase, interfaces, context, missing_feature, coverage, released
+
+
+@coverage.basic
+@released(
+    dotnet="?", golang="?", java="?", nodejs="?", php_appsec="?", python="?", ruby="?", cpp="?",
+)
+class TestIastNoRequest(BaseTestCase):
+    """ IAST deals with vulnerabilities outside of request """
+
+    EXPECTATIONS = {"java": {"LOCATION": "com.datadoghq.system_tests.springboot.iast.utils.CryptoExamples"}}
+
+    def __expected_location(self):
+        expected = self.EXPECTATIONS.get(context.library.library)
+        return expected.get("LOCATION") if expected else None
+
+    @missing_feature(reason="Need to implement span creation")
+    def test_span_creation(self):
+        """ IAST creates a new span when no request is active """
+
+        def iast_created_span_validator(item):
+            return item["span"].get("name") == "iast.vulnerability"
+
+        interfaces.library.expect_iast_vulnerabilities(
+            vulnerability_count=1,
+            location_path=self.__expected_location(),
+            vulnerability_type="WEAK_HASH",
+            span_validator=iast_created_span_validator,
```

Comment:
If i'm not wrong, `span_validator` is executed on all spans triggered by the request, whatever it contains an iast event or not.

If yes, IMO, it would be simpler to use  `interfaces.library.add_span_validation(r, iast_created_span_validator)` , rather than duplicating  complexity inside `_BaseAppSecIastValidation`.

Reply:
The reason I did it like this is because the span validator is only applied if the vulnerabilities expectation matches, but while doing it I also consider what you mention. 

I will remove the extra complexity since we can survive without it

Reply:
@cbeauchesne I've updated the PR to simplify everything, let me know what you think